app/src/main/java/io/neurolab/model/FileOutputTask.java

```diff
@@ -0,0 +1,85 @@
+package io.neurolab.model;
+
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+
+import io.neurolab.main.NFBServer;
+
+public class FileOutputTask implements Task {
+
+    private NFBServer nfbServer;
+    private PrintWriter out = null;
+    private byte[] stopBytes;
+    private String fileName = "defaultFileOutput.bw";
+    private boolean write = false;
+
+    public FileOutputTask(NFBServer nfbServer) {
+        this.nfbServer = nfbServer;
+        this.stopBytes = doubleToByteArray(Double.MAX_VALUE);
+    }
+
+    public void setFileName(String fileName) {
+        this.fileName = fileName;
+        init();
+    }
+
+    public void setWrite(boolean write) {
+        this.write = write;
+    }
+
+    @Override
+    public void run() {
+        if ((write) && (!nfbServer.getInputData().isEmpty())) {
+            try {
+                for (int s = 0; s < nfbServer.getNumSamples(); s++) {
+                    double[] currentPacket = nfbServer.getInputData().get(s);
+
+                    out.print(nfbServer.getCurrentTimestamp() + ",");
+                    for (int c = 0; c < currentPacket.length; c++)
+                        out.print(currentPacket[c] + ",");
+                    out.println("");
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
```

Comment:
@jddeep Remove an extra line of space here.

Reply:
Thanks @suresh-gandhi . Done

Reply:
Yeah @suresh-gandhi done! Thanks.